editor/project_manager.cpp

```diff
@@ -69,270 +70,303 @@ constexpr int GODOT4_CONFIG_VERSION = 5;
-			get_ok_button()->set_disabled(true);
-			return "";
-		}
+			if (target_path.get_base_dir().is_empty() || !d->dir_exists(target_path.get_base_dir())) {
+				_set_message(TTR("The parent directory of the path specified doesn't exist."), MESSAGE_ERROR, target_path_input_type);
+				return "";
+			}
 
-	} else {
-		// Check if the specified folder is empty, even though this is not an error, it is good to check here.
-		d->list_dir_begin();
-		is_folder_empty = true;
-		String n = d->get_next();
-		while (!n.is_empty()) {
-			if (!n.begins_with(".")) {
-				// Allow `.`, `..` (reserved current/parent folder names)
-				// and hidden files/folders to be present.
-				// For instance, this lets users initialize a Git repository
-				// and still be able to create a project in the directory afterwards.
-				is_folder_empty = false;
-				break;
+			// We already know the parent directory exists and is valid, so we only need to check the last part.
+			if (target_path.get_file() != OS::get_singleton()->get_safe_dir_name(target_path.get_file())) {
+				_set_message(TTR("The path specified contains invalid characters."), MESSAGE_ERROR, target_path_input_type);
+				return "";
 			}
-			n = d->get_next();
-		}
-		d->list_dir_end();
 
-		if (!is_folder_empty) {
-			if (valid_path == OS::get_singleton()->get_environment("HOME") || valid_path == OS::get_singleton()->get_system_dir(OS::SYSTEM_DIR_DOCUMENTS) || valid_path == OS::get_singleton()->get_executable_path().get_base_dir()) {
-				_set_message(TTR("You cannot save a project in the selected path. Please make a new folder or choose a new path."), MESSAGE_ERROR);
-				get_ok_button()->set_disabled(true);
+			if (d->dir_exists(target_path)) {
+				// The path is not necessarily empty here, but we will update the message later if it isn't.
+				_set_message(TTR("The project folder already exists and is empty."), MESSAGE_SUCCESS, target_path_input_type);
+			} else {
+				_set_message(TTR("The project folder will be automatically created."), MESSAGE_SUCCESS, target_path_input_type);
+			}
+		} else {
+			if (target_path.is_empty() || target_path.is_relative_path() || !d->dir_exists(target_path)) {
+				_set_message(TTR("The path specified doesn't exist."), MESSAGE_ERROR, target_path_input_type);
 				return "";
 			}
 
-			_set_message(TTR("The selected path is not empty. Choosing an empty folder is highly recommended."), MESSAGE_WARNING);
-			get_ok_button()->set_disabled(false);
-			return valid_path;
+			// The path is not necessarily empty here, but we will update the message later if it isn't.
+			_set_message(TTR("The project folder exists and is empty."), MESSAGE_SUCCESS, target_path_input_type);
+		}
+
+		if (d->change_dir(target_path) == OK) {
+			// Check if the directory is empty. Not an error, but we want to warn the user.
+			d->list_dir_begin();
+			String n = d->get_next();
+			while (!n.is_empty()) {
+				if (!n.begins_with(".")) {
```

Comment:
```suggestion
				if (n[0] != '.') {
```
Known to be non-empty so can use faster check